src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -201,6 +204,22 @@ public void CanBindIConfigurationSectionWithDerivedOptionsSection()
             Assert.Null(options.Section.Value);
         }
 
+        [Fact]
+        public void CanBindConfigurationKeyNameAttributes()
+        {
+            var dic = new Dictionary<string, string>
+            {
+                {"Named_Property", "Yo"},
+            };
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dic);
+            var config = configurationBuilder.Build();
+
+            var options = config.Get<ComplexOptions>();
+            
+            Assert.Equal("Yo", options.NamedProperty);
+        }
+
```

Comment:
For completeness, maybe have a test for both field and properties since the attribute supports both?

Reply:
Have the project still open. :)

Reply:
What issues are you hitting with Fields?

Reply:
I don't think we bound to fields before did we?  That's sorta why I asked

Reply:
You are right. Then we should drop the field support from the attribute

Reply:
Removed.